torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -505,9 +505,62 @@ Tensor masked_fill_backward(const Tensor& grad, const Tensor& mask) {
       : grad.masked_select(mask).sum();
 }
 
-Tensor mul_tensor_backward(Tensor grad, Tensor other, ScalarType self_st) {
-  auto out = grad * other.conj();
-  return handle_r_to_c(self_st, out);
+Tensor mul_tensor_backward(
+    const Tensor& grad,
+    const Tensor& other,
+    bool self_is_sparse,
```

Comment:
To facilitate supporting sparse compressed tensors in `mul_tensor_backward`, could we change this interface to:
```
Tensor mul_tensor_backward(
￼    const Tensor& grad,
￼    const Tensor& other,
     Layout self_layout,
     ...
```
or
```C++
Tensor mul_tensor_backward(
￼    const Tensor& grad,
￼    const Tensor& other,
     const Tensor& self,
     ...
```
so that `mul_tensor_backward` implementation could dispatch to the proper handling function based on `self.layout()`?

Reply:
Nope, that will not work like that. `mul_tensor_backward` is an implicit function.